src/App.tsx

```diff
@@ -14,11 +14,163 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods : string[];
+};
+
+function showHeading(goodsArray : string[]) {
+  if (goodsArray.length === 0) {
+    return ('No goods selected');
+  }
+
+  if (goodsArray.length === 1) {
+    return (`${goodsArray} is selected`);
+  }
+
+  let newHeading = '';
+
+  if (goodsArray.length > 1) {
+    for (let i = 0; i <= goodsArray.length - 2; i += 1) {
+      if ((i + 2) !== goodsArray.length) {
+        newHeading += `${goodsArray[i]}, `;
+      } else {
+        newHeading += `${goodsArray[i]} and ${goodsArray[i + 1]}`;
+      }
+    }
+
+    newHeading += ' are selected';
+  }
+
+  return newHeading;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          { showHeading(this.state.selectedGoods) }
+        </h1>
+        <div className="goods">
+          <ul className="goods__list">
+            {goodsFromServer.map(el => (
+              <li
+                className={
+                  (this.state.selectedGoods.includes(el)
+                    ? 'goods__list-item-text-selected goods__list-item'
+                    : 'goods__list-item')
+                }
+                key={el}
+              >
+                {el}
+                <button
+                  type="button"
+                  onClick={() => {
```

Comment:
Will be better to create separate function in your class with this logic and add only link on this function in the jsx